src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +65,61 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
             return this.configurationFactory.CreateAnyCallConfiguration(manager, rule);
         }
 
+        public IVoidArgumentValidationConfiguration CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
+            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+
+            var setterExpression = BuildSetterFromGetter(propertySpecification);
+            var fake = this.expressionParser.GetFakeManagerCallIsMadeOn(setterExpression);
+            var rule = this.ruleFactory.Invoke(setterExpression);
+
+            fake.AddRuleFirst(rule);
+
+            return this.configurationFactory.CreateConfiguration(fake, rule);
+        }
+
+        private string GetExpressionDescription<TValue>(Expression<Func<TValue>> expression)
+        {
+            var matcher = new ExpressionCallMatcher(
+                expression,
+                ServiceLocator.Current.Resolve<ExpressionArgumentConstraintFactory>(),
+                ServiceLocator.Current.Resolve<MethodInfoManager>(),
+                this.callExpressionParser);
+
+            var expressionDescription = matcher.DescriptionOfMatchingCall;
+            return expressionDescription;
+        }
+
+        private LambdaExpression BuildSetterFromGetter<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            MemberExpression memberExpression = propertySpecification.Body as MemberExpression;
+            if (memberExpression == null)

```

Comment:
Yeah, I added a test for an indexer and discovered this very thing this (Thursday, as I type) morning. I started adding the conversion from indexed get to indexed set, but am not quite there (having problems finding the instance).
Also, I did not know that VB supported named indexers. I guess in that case the best way to detect them would be to look for `get_TheProperty` and assume nobody made a method that's so named… Which I've never tried, but it must be possible.